library/src/device/generator/generator.pass.hpp

```diff
@@ -844,31 +844,19 @@ namespace StockhamGenerator
                             }
                             else
                             {
+                                passStr += rType + "wx, wy, rx, ry;\n\t\t";
+                                passStr += "wx = W.x; wy = W.y;\n\t\t";
+                                passStr += "rx = " + regRealIndex + "; ry = " + regImagIndex
+                                           + ";\n\t\t";
                                 if(fwd)
                                 {
-                                    passStr += "TR = (W.x * ";
-                                    passStr += regRealIndex;
-                                    passStr += ") - (W.y * ";
-                                    passStr += regImagIndex;
-                                    passStr += ");\n\t\t";
-                                    passStr += "TI = (W.y * ";
-                                    passStr += regRealIndex;
-                                    passStr += ") + (W.x * ";
-                                    passStr += regImagIndex;
-                                    passStr += ");\n\t\t";
+                                    passStr += "TR = wx * rx - wy * ry;\n\t\t";
```

Comment:
You are just rearranging expressions so the compiler bug is avoided?

Reply:
There's a local variable that serves as an explicit cast.